### PR TITLE
Feat/create project

### PR DESCRIPTION
**Pull request by onegoel**
Added functionality of projects and members management.

User can create a project and become its admin. Admin has privileges to control project and perform member management. Roles are assigned to project members only by the project's admin.

CHANGES TO BACKEND DESIGN

- created a new DB for lookup of meta data, called Management DB
- contains Project, Member, ProjectMember models
- Project (projectId*, projectName, projectDesc) - projectId will be JIRA code of project
- Member (memberId* + meta data of member, subject to future enhancements)
- ProjectMember (memberId*, projectId*, email - unique email+projectId for easy lookup)
- added Role enum (ADMIN, LEADER, MEMBER)

- existing DB is now referred to as Dashboard DB
- added a column in each main model of every feature called projectId, used to link feature data to project

FEATURES

1. Create new project - user can create a new project and assume role of ADMIN
2. List projects by current user email - lists project current user (req.user.email) is part of

> Role validation middleware - protects the following routes for only ADMIN to see
1. Add new project member
2. Update member role
3. Remove project member
4. Update project info
5. Delete project

> Member validation middleware - protects the following routes only for project members to see
1. Get all details of a project by project id
2. Get all members involved in a project by project id

Also includes CRUD endpoints for Member model, parked for future enhancements.

MISCELLANEOUS
- generated separate Prisma clients for both DB's
- added an updated example .env
- body validation where necessary
- fully integrated with okta auth

Note: Please skip over the migration.sql files bcoz these are auto-generated by Prisma. For a brief overview of the current DB schemas, please go to the schema.prisma files in prism/management and prisma/dashboard folders. May also ignore the src/generated folder files, those are auto generated Prisma clients. Thanks.

### CODE REVIEW

**Comment by onegoel** [comment]
NEW CHANGES

DB
- added memberId column to PO notes table
- soft deletes for Project and Member (for future enhancements)

MIDDLEWARES 
- role validation: checks if admin
- member validation: checks if member
- middlewares store role and memberId in req.user

PO NOTES
- refactored backend for PO notes
- routes now include projectId
- middlewares ensure the following
- only ADMIN can create, edit and delete PO notes, throws a 403 otherwise
- when ADMIN creates PO note, memberId is taken from req.user and stored in db
- only project members can see PO notes for that project
- non project members not authorised to access PO notes for a project

File: src/generated/management/index.js [hunk 1]
```diff
@@ -0,0 +1,204 @@
+
```
Comment (sugumar-mani) [inline src/generated/management/index.js:1]: Do we need to commit generated files

Comment (onegoel) [reply src/generated/management/index.js:1]: No I will remove that. We will need empty folders at src/generated however so we can write a script for that.

File: src/controllers/management/management.controller.js [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+      email,
+      name
+    );
+    res.status(201).json({
+      message: 'Successfully created new project.', 
+      result
+    });
+  }
+  catch (er) {
+    console.log(er);
```
Comment (sugumar-mani) [inline src/controllers/management/management.controller.js:33]: remove if not needed

**Review by sugumar-mani — APPROVED** [review]

**Review by sugumar-mani — APPROVED** [review]
Please resolve merge conflicts before merge

**Review by sugumar-mani — APPROVED** [review]

**Pull request merged**